mmedit/models/video_interpolater/basic_interpolater.py

```diff
@@ -0,0 +1,308 @@
+                Default: None.
+            meta (list[dict]): Meta data, such as path of target file.
+                Default: None.
+            save_image (bool): Whether to save image. Default: False.
+            save_path (str): Path to save image. Default: None.
+            iteration (int): Iteration for the saving image name.
+                Default: None.
+
+        Returns:
+            dict: Output results.
+        """
+        output = self.generator(inputs)
+        if self.test_cfg is not None and self.test_cfg.get('metrics', None):
+            assert target is not None, (
+                'evaluation with metrics must have target images.')
+            results = dict(eval_result=self.evaluate(output, target))
+        else:
+            results = dict(inputs=inputs.cpu(), output=output.cpu())
+            if target is not None:
+                results['target'] = target.cpu()
+
+        # save image
+        if save_image:
+            self._save_image(meta, iteration, save_path, output)
+
+        return results
+
+    @staticmethod
+    def _save_image(meta, iteration, save_path, output):
+        """Save the image.
+
+        Args:
+            meta (list[dict]): Meta data, such as path of target file.
+                Default: None. These dictionaries should contain
+                'target_path' (str of a path) or 'inputs_path' (list of str)
+            iteration (int): Iteration for the saving image name.
+                Default: None.
+            save_path (str): Path to save image. Default: None.
+            output (Tensor): Output image.
+        """
+
+        if output.ndim == 4:  # an image
+            img_name = meta[0]['key'].replace('/', '_')
+            if isinstance(iteration, numbers.Number):
+                save_path = osp.join(save_path,
+                                     f'{img_name}-{iteration + 1:06d}.png')
+            elif iteration is None:
+                save_path = osp.join(save_path, f'{img_name}.png')
+            else:
+                raise ValueError('iteration should be number or None, '
+                                 f'but got {type(iteration)}')
+            mmcv.imwrite(tensor2img(output), save_path)
+        elif output.ndim == 5:  # a sequence
+            folder_name = meta[0]['key'].split('/')[0]
+            for i in range(0, output.size(1)):
+                if isinstance(iteration, numbers.Number):
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}-{iteration + 1:06d}.png')
+                elif iteration is None:
+                    save_path_i = osp.join(save_path, folder_name,
+                                           f'{i:08d}.png')
+                else:
+                    raise ValueError('iteration should be number or None, '
+                                     f'but got {type(iteration)}')
+                mmcv.imwrite(tensor2img(output[:, i, :, :, :]), save_path_i)
```

Comment:
As discussed:

1. If the output has only 4 dimensions, then the convention is the middle frame. Then we can use `00000000_050.png`.

2. If the output has 5 dimensions, we assume they are multiple intermediate frames between 2 frames. For example, in CAIN, the output would be `torch.stack([x0.25, 0.5, 0.75], dim=1)`. Then we can use something like `00000000_025.png`, `00000000_050.png`, `00000000_075.png`, assuming x0 has name `00000000.png`.